Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -615,12 +637,47 @@ void CGPathAddRoundedRect(
     UNIMPLEMENTED();
 }
 
+int _CGPathPointCountForElementType(CGPathElementType type) {
+    int pointCount = 0;
+
+    switch (type) {
+        case kCGPathElementMoveToPoint:
+        case kCGPathElementAddLineToPoint:
+            pointCount = 1;
+            break;
+        case kCGPathElementAddQuadCurveToPoint:
+            pointCount = 2;
+            break;
+        case kCGPathElementAddCurveToPoint:
+            pointCount = 3;
+            break;
+        case kCGPathElementCloseSubpath:
+            pointCount = 0;
+            break;
+    }
+    return pointCount;
+}
+
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 void CGPathApply(CGPathRef path, void* info, CGPathApplierFunction function) {
-    UNIMPLEMENTED();
+    if (path == NULL) {
+        return;
+    }
+
```

Comment:
Nit: it looks like CGPathApply is the only one that is doing the null check.  fail fast (by not checking NULL) will make it consistent.